controllers/progressivesync_controller.go

```diff
@@ -267,6 +289,41 @@ func (r *ProgressiveSyncReconciler) getClustersFromSelector(ctx context.Context,
 	return secrets, nil
 }
 
+// calculateHashValue returns a hash value string for the ApplicationSet spec
+func (r *ProgressiveSyncReconciler) calculateHashedSpec(ctx context.Context, ps *syncv1alpha1.ProgressiveSync) (string, error) {
+	key := client.ObjectKeyFromObject(ps)
+	latest := syncv1alpha1.ProgressiveSync{}
+	if err := r.Client.Get(ctx, key, &latest); err != nil {
+		r.Log.Error(err, "Failed to retrieve the ProgressiveSync object while calculating the hash value")
+		return "", err
+	}
+
+	apiMetadata := strings.Split(*latest.Spec.SourceRef.APIGroup, "/")
+	genericObj := unstructured.Unstructured{}
+	genericObj.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   apiMetadata[0],
+		Kind:    latest.Spec.SourceRef.Kind,
+		Version: apiMetadata[1],
+	})
+	if err := r.Client.Get(ctx, client.ObjectKey{Namespace: "argocd", Name: latest.Spec.SourceRef.Name}, &genericObj); err != nil {
+		r.Log.Error(err, "Failed to retrieve the unstruct object")
+		return "", err
+	}
+
+	objectToHash, ok := genericObj.UnstructuredContent()["spec"]
+	if !ok {
+		err := errors.New("`spec` missing from generic object")
+		r.Log.Error(err, "Missing value from map")
+
+		return "", err
+	}
+	hashValue := fnv.New32a()
+	hash.DeepHashObject(hashValue, objectToHash)
+	r.Log.Info("Successfully calculate the hash value of the Spec")
```

Comment:
Maybe this should be a debug log line instead of info?

Reply:
Not sure I am missing something, but the logger we currently use doesn't have a Debug level of logging - https://pkg.go.dev/github.com/go-logr/logr#pkg-index. There is the idea of levels of verbosity that is set with numbers, but I am not quite sure how this translates to what we want to achieve 🤔

Reply:
You are correct, and we have a ticket to tackle the logging at https://github.com/Skyscanner/applicationset-progressive-sync/issues/115

Reply:
Ok, thanks, I will resolve as we have an issue to track this.